openmdao/core/problem.py

```diff
@@ -431,6 +436,11 @@ def final_setup(self):
             self._setup_status = 2
             self._set_initial_conditions()
 
+        # check for post-setup hook
+        if Problem._post_setup_func is not None:
+            Problem._post_setup_func(self)
+            sys.exit(0)
```

Comment:
would it be more general for the post_setup function to decide whether it's appropriate to exit or not?

Reply:
Agreed.  I had actually just made that change before I saw your comment.